qxf2_scheduler/routes.py

```diff
@@ -65,103 +69,177 @@ def index():
+                parsed_interviewer_details[0]['time'] = [
+                    parsed_interviewer_details[0]['time'], parsed_interviewer_detail['time']]
+
+    return parsed_interviewer_details
+
+
+@app.route("/<interviewer_id>/interviewer/")
+def read_interviewer_details(interviewer_id):
+    "Displays all the interviewer details"
+    # Fetching the Interviewer detail by joining the Interviewertimeslots tables and Interviewer tables
+    interviewer_details = Interviewers.query.join(Interviewertimeslots, Interviewers.interviewer_id == Interviewertimeslots.interviewer_id).filter(
+        Interviewers.interviewer_id == interviewer_id).values(Interviewers.interviewer_name, Interviewers.interviewer_email, Interviewers.interviewer_designation, Interviewers.interviewer_id, Interviewertimeslots.interviewer_start_time, Interviewertimeslots.interviewer_end_time)
+
+    parsed_interviewer_details = form_interviewer_details(interviewer_details)
+
+    return render_template("read-interviewers.html", result=parsed_interviewer_details)
+
+
+@app.route("/<interviewer_id>/interviewer/edit/", methods=['GET', 'POST'])
+def edit_interviewer(interviewer_id):
+    "Edit the interviewers"    
+    # This query fetch the interviewer details by joining the time slots table and interviewers table.
+    edit_interviewer_details = Interviewers.query.join(Interviewertimeslots, Interviewers.interviewer_id == Interviewertimeslots.interviewer_id).values(
+        Interviewers.interviewer_id, Interviewers.interviewer_name, Interviewers.interviewer_email, Interviewers.interviewer_designation, Interviewertimeslots.interviewer_start_time, Interviewertimeslots.interviewer_end_time)
     
+    parsed_interviewer_details = form_interviewer_details(edit_interviewer_details)
+
+    if request.method == "POST":
+        interviewer_name = request.form.get('name')
+        data = {'interviewer_name': interviewer_name}
+        print("ia interviewer is", interviewer_name, file=sys.stderr)
+        edit_interviewers = Interviewers.query.filter(Interviewers.interviewer_id == interviewer_id).update({'interviewer_name': request.form.get(
+            'name'), 'interviewer_email': request.form.get('email'), 'interviewer_designation': request.form.get('designation')})
+        db.session.commit()
+
+        # Adding the time slots in the interviewerstimeslots table
+        interviewer_time_slots = eval(request.form.get('timeObject'))
+        interviewer_start_time = interviewer_time_slots['starttime']
+        interviewer_end_time = interviewer_time_slots['endtime']
+
+        # Updating the table based on the time id of the interviewertimeslots table
+        # Fetching the time ids of interviewer
+        total_rows_of_interviewer_in_time_table = Interviewertimeslots.query.filter(
+            Interviewertimeslots.interviewer_id == interviewer_id).values(Interviewertimeslots.time_id)
+
+        # Updating the timeslots
+        i = 0
```

Comment:
Hmmm ... I am not sure this is correct. Aren't you assuming that the number of chunks will be the same? What happens if my time slots were:
a) Initially (11,13) and (14,20) 
b) I change to (9,12) and (13,15) and (17,20)

With this logic, I think you will end up missing the (17,20) slot in your logic